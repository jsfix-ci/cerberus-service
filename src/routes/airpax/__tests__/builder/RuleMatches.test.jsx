import React from 'react';
import renderer from 'react-test-renderer';

import RulesMatches from '../../TaskDetails/builder/RuleMatches';
import taskDetailsSelectors from '../../__fixtures__/taskData_AirPax_TaskDetails_Selectors.fixture.json';

describe('RuleMatches', () => {
  it('should render RuleMatches component', () => {
    const tree = renderer.create(<RulesMatches version={taskDetailsSelectors.versions[0]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
