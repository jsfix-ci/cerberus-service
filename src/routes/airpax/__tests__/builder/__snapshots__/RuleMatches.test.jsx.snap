// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuleMatches should render RuleMatches component 1`] = `
<div>
  <h2
    className="govuk-heading-m header-bg-grey"
  >
    Rules matched
  </h2>
  <div
    className="govuk-grid-row"
  >
    <div
      className="govuk-grid-column-one-quarter"
    >
      <h4
        className="govuk-heading-s"
      >
        Rule name
      </h4>
      <p>
        Selector Matched Rule
      </p>
    </div>
    <div
      className="govuk-grid-column-one-quarter"
    >
      <h4
        className="govuk-heading-s"
      >
        Threat
      </h4>
      <p>
        Tier 1
      </p>
    </div>
    <div
      className="govuk-grid-column-one-quarter"
    >
      <h4
        className="govuk-heading-s"
      >
        Rule Version
      </h4>
      <p>
        1
      </p>
    </div>
    <div
      className="govuk-grid-column-one-quarter"
    >
      <h4
        className="govuk-heading-s"
      >
        Abuse Type
      </h4>
      <p>
        National Security at the Border
      </p>
    </div>
  </div>
  <div
    className="govuk-grid-row"
  >
    <div
      className="govuk-grid-column-one-quarter false"
    >
      <h4
        className="govuk-heading-s"
      >
        Description
      </h4>
      <p>
        Test
      </p>
    </div>
    <div
      className="govuk-grid-column-one-quarter float-right"
    >
      <h4
        className="govuk-heading-s"
      >
        Agency
      </h4>
      <p>
        
      </p>
    </div>
  </div>
  <div
    className="govuk-grid-row"
  >
    <div
      className="govuk-grid-column-full govuk-!-margin-bottom-6"
    >
      <h4
        className="govuk-heading-s govuk-!-margin-bottom-2"
      >
        Risk indicators (
        2
        )
      </h4>
      <table
        className="hods-table"
        role="table"
      >
        <thead
          className="hods-table__heading"
        >
          <tr
            role="row"
          >
            <th
              role="columnheader"
              scope="col"
            >
              Entity type
            </th>
            <th
              role="columnheader"
              scope="col"
            >
              Attribute
            </th>
            <th
              role="columnheader"
              scope="col"
            >
              Operator
            </th>
            <th
              role="columnheader"
              scope="col"
            >
              Value(s)
            </th>
          </tr>
        </thead>
        <tbody
          className="hods-table__body"
        >
          <tr
            className="hods-table__row"
            role="row"
          >
            <td
              className="hods-table__cell"
              role="cell"
            >
              <span
                className="hods-table__heading"
              >
                Entity type
              </span>
              Message
            </td>
            <td
              className="hods-table__cell"
              role="cell"
            >
              <span
                className="hods-table__heading"
              >
                Attribute
              </span>
              mode
            </td>
            <td
              className="hods-table__cell"
              role="cell"
            >
              <span
                className="hods-table__heading"
              >
                Operator
              </span>
              in
            </td>
            <td
              className="hods-table__cell"
              role="cell"
            >
              <span
                className="hods-table__heading"
              >
                Value(s)
              </span>
              [RORO Accompanied Freight, RORO Tourist, RORO Unaccompanied Freight]
            </td>
          </tr>
          <tr
            className="hods-table__row"
            role="row"
          >
            <td
              className="hods-table__cell"
              role="cell"
            >
              <span
                className="hods-table__heading"
              >
                Entity type
              </span>
              Message
            </td>
            <td
              className="hods-table__cell"
              role="cell"
            >
              <span
                className="hods-table__heading"
              >
                Attribute
              </span>
              selectorsMatched
            </td>
            <td
              className="hods-table__cell"
              role="cell"
            >
              <span
                className="hods-table__heading"
              >
                Operator
              </span>
              equal
            </td>
            <td
              className="hods-table__cell"
              role="cell"
            >
              <span
                className="hods-table__heading"
              >
                Value(s)
              </span>
              true
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div>
    <h2
      className="govuk-heading-m header-bg-grey"
    >
      Other rule matches (
      1
      )
    </h2>
    <div>
      <div
        className="govuk-grid-row"
      >
        <div
          className="govuk-grid-column-one-quarter"
        >
          <h4
            className="govuk-heading-s"
          >
            Rule name
          </h4>
          <p>
            Checkin lead time
          </p>
        </div>
        <div
          className="govuk-grid-column-one-quarter"
        >
          <h4
            className="govuk-heading-s"
          >
            Threat
          </h4>
          <p>
            Tier 4
          </p>
        </div>
        <div
          className="govuk-grid-column-one-quarter"
        >
          <h4
            className="govuk-heading-s"
          >
            Rule Version
          </h4>
          <p>
            1
          </p>
        </div>
        <div
          className="govuk-grid-column-one-quarter"
        >
          <h4
            className="govuk-heading-s"
          >
            Abuse Type
          </h4>
          <p>
            Class A Drugs
          </p>
        </div>
      </div>
      <details
        className="govuk-details"
        data-module="govuk-details"
      >
        <summary
          className="govuk-details__summary"
        >
          <span
            className="govuk-details__summary-text"
          >
            View further details
          </span>
        </summary>
        <div
          className="govuk-details__text"
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <div
            className="govuk-grid-row govuk-!-margin-left-1"
          >
            <div
              className="govuk-grid-column-one-quarter false"
            >
              <h4
                className="govuk-heading-s"
              >
                Description
              </h4>
              <p>
                Test
              </p>
            </div>
            <div
              className="govuk-grid-column-one-quarter float-right"
            >
              <h4
                className="govuk-heading-s"
              >
                Agency
              </h4>
              <p>
                
              </p>
            </div>
          </div>
          <div
            className="govuk-grid-column-full"
          >
            <h4
              className="govuk-heading-s"
            >
              Risk indicators (
              2
              )
            </h4>
            <table
              className="hods-table"
              role="table"
            >
              <thead
                className="hods-table__heading"
              >
                <tr
                  role="row"
                >
                  <th
                    role="columnheader"
                    scope="col"
                  >
                    Entity type
                  </th>
                  <th
                    role="columnheader"
                    scope="col"
                  >
                    Attribute
                  </th>
                  <th
                    role="columnheader"
                    scope="col"
                  >
                    Operator
                  </th>
                  <th
                    role="columnheader"
                    scope="col"
                  >
                    Value(s)
                  </th>
                </tr>
              </thead>
              <tbody
                className="hods-table__body"
              >
                <tr
                  className="hods-table__row"
                  role="row"
                >
                  <td
                    className="hods-table__cell"
                    role="cell"
                  >
                    <span
                      className="hods-table__heading"
                    >
                      Entity type
                    </span>
                    Booking
                  </td>
                  <td
                    className="hods-table__cell"
                    role="cell"
                  >
                    <span
                      className="hods-table__heading"
                    >
                      Attribute
                    </span>
                    checkInLeadTime
                  </td>
                  <td
                    className="hods-table__cell"
                    role="cell"
                  >
                    <span
                      className="hods-table__heading"
                    >
                      Operator
                    </span>
                    between
                  </td>
                  <td
                    className="hods-table__cell"
                    role="cell"
                  >
                    <span
                      className="hods-table__heading"
                    >
                      Value(s)
                    </span>
                    [1, 24]
                  </td>
                </tr>
                <tr
                  className="hods-table__row"
                  role="row"
                >
                  <td
                    className="hods-table__cell"
                    role="cell"
                  >
                    <span
                      className="hods-table__heading"
                    >
                      Entity type
                    </span>
                    Message
                  </td>
                  <td
                    className="hods-table__cell"
                    role="cell"
                  >
                    <span
                      className="hods-table__heading"
                    >
                      Attribute
                    </span>
                    mode
                  </td>
                  <td
                    className="hods-table__cell"
                    role="cell"
                  >
                    <span
                      className="hods-table__heading"
                    >
                      Operator
                    </span>
                    in
                  </td>
                  <td
                    className="hods-table__cell"
                    role="cell"
                  >
                    <span
                      className="hods-table__heading"
                    >
                      Value(s)
                    </span>
                    [RORO Accompanied Freight, RORO Tourist, RORO Unaccompanied Freight]
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </details>
    </div>
  </div>
</div>
`;
